### PR TITLE
allow QP on re-writes

### PR DESCRIPTION
**Pull request by jackstine**
# Description

Need to allow Query Parameters on Re-writes. Re-writes will have a different route validation than other route matches. The reason is that customers use prefix re-writes to perform query parameters.

# Context

Users can not perform query parameter transformations on URLs with currnet route validation.

# Checklist:

- [ ] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [ ] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [ ] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [ ] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: projects/gloo/pkg/translator/route_config.go [hunk 1]
```diff
@@ -842,6 +843,12 @@ func validatePath(path, name string, routeReport *validationapi.RouteReport) {
 	}
 }
 
+func validatePrefixRewrite(rewrite, name string, routeReport *validationapi.RouteReport) {
+	if _, err := url.Parse(rewrite); err != nil {
+		validation.AppendRouteError(routeReport, validationapi.RouteReport_Error_ProcessingError, errors.Wrapf(err, "the rewrite is invalid: %s", rewrite).Error(), name)
+	}
```
Comment (kdorosh) [inline projects/gloo/pkg/translator/route_config.go:849]: https://pkg.go.dev/net/url#URL

we can also use the result and validate `Url.Path` with `validatePath()` here

File: projects/gloo/pkg/translator/route_config.go [hunk 2]
```diff
@@ -230,11 +231,11 @@ func validateEnvoyRoute(r *envoy_config_route_v3.Route, routeReport *validationa
 	validatePath(match.GetPath(), name, routeReport)
 	validatePath(match.GetPrefix(), name, routeReport)
 	validatePath(match.GetPathSeparatedPrefix(), name, routeReport)
-	validatePath(route.GetPrefixRewrite(), name, routeReport)
-	validatePath(re.GetPrefixRewrite(), name, routeReport)
```
Comment (kdorosh) [inline projects/gloo/pkg/translator/route_config.go:234]: would like to see unit test update that allows query params in prefix rewrite

File: projects/gloo/pkg/translator/route_config.go [hunk 3]
```diff
@@ -842,6 +843,12 @@ func validatePath(path, name string, routeReport *validationapi.RouteReport) {
 	}
 }
 
+func validatePrefixRewrite(rewrite, name string, routeReport *validationapi.RouteReport) {
```
Comment (saiskee) [inline projects/gloo/pkg/translator/route_config.go:846]: Can we add a small unit test for this method that makes sure it allows QPs and that ValidatePath doesn't allow QPs.

File: projects/gloo/pkg/translator/route_config.go (hunk 1 above)
Comment (jackstine) [reply projects/gloo/pkg/translator/route_config.go:849]: that way we only validate the path, and not the QP portion. Like it

File: changelog/v1.14.0-beta8/allow-query-parameters-on-re-writes.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+changelog:
+- type: FIX
+  issueLink: https://github.com/solo-io/gloo/issues/7715#issuecomment-1410014270
+  resolvesIssue: false
+  description: >-
+    allow query parameters on re-writes
```
Comment (nfuden) [inline changelog/v1.14.0-beta8/allow-query-parameters-on-re-writes.yaml:6]: nit: Can we make this more descriptive for customers?

Comment (jackstine) [reply changelog/v1.14.0-beta8/allow-query-parameters-on-re-writes.yaml:6]: like fixing issues with validation?

Comment (nfuden) [reply changelog/v1.14.0-beta8/allow-query-parameters-on-re-writes.yaml:6]: Yeah
```suggestion
    No longer fail validation when specifying query parameters on re-writes.
```

**Review by nfuden — APPROVED** [review]

**Comment by omerlh** [comment]
Thank you!

**Review by saiskee — APPROVED** [review]

**Review by kdorosh — APPROVED** [review]

**Pull request merged**